design/hardware-classification-controller/support-for-new-parameters-hwcc.md

```diff
@@ -0,0 +1,268 @@
+   Types of architecture are 32-bit (x86) and 64-bit
+   (x86-64, IA64, and AMD64).
+
+   ```yaml
+   hardware:
+       cpu:
+          arch: "x86_64"
+   ```
+
+1. Disk
+
+   `type` will come under disk parameter.
+
+   User can provide multiple disk type (RAID/HDD/SSD/NVMe), count and
+   individual size based on requirement.
+
+   Investigation Details: Disk type can be identified using combination
+   of rotational and hctl parameters.
+   Here `hctl` represents :
+
+   1. SCSI adapter number `host`
+   1. channel number `bus`
+   1. id number `target`
+   1. number of logical units `lun`
+
+   and rotational value will be true if disk is HDD and false
+   represents individual SSD, also software RAID can be SSD RAID or
+   HDD RAID with rotational value true.
+
+   We are taking example of Dell hardware here.
+
+   1. Individual HDD/SSD : with `hctl` as 0:0:N:0 and rotational flag as
+   True/False.
+   1. PERC RAID of HDDs/SSDs : with `hctl` as 0:N:0:0/0:N:N:0 and rotational
+   flag as True.
+   1. Dell BOSS Controller Individual SSDs : with `hctl` as N:0:0:0 and
+   rotational flag as False.
+   1. Dell BOSS Controller Virtual Disk (RAID) : with `hctl` as N:0:0:0 and
+   rotational flag as True.
+   1. NVMe : No `hctl` Pattern, rotational flag as False and model name
+   contains NVMe keyword.
+
+   Below is sample introspection data for Disk.
+
+     ```yaml
+     hardware:
+        disks:
+          - name: "/dev/sda"
+            model: "KPM5XVUG3T84"
+            rotational: "true"
+            hctl: "0:0:N:0"
+     ```
+
+1. NIC
+
+   Adding new parameter `vendorName` under NIC. User can provide NIC
+   requirement in the form of `vendor`.
+
+   While classification of `vendor` parameter, use sub-string matching
+   will be done.
+
+   Investigation details: Below is sample introspection data for NICs.
+   `model` field under `nics` contains combination of Vendor ID and
+   Product ID.
+
+     ```yaml
+     hardware:
+       nics:
+       - ip: ""
+         mac: 24:6e:96:aa:bb:cc
+         model: 0x8086 0x1234
+         name: eth3
+         pxe: false
+     ```
+
+1. Firmware
+
+   Firmware is new parameter to be introduced in HardwareClassification
+   which contains BIOS information. User can specify BIOS requirements
+   by providing `vendor` and `version` details.
+
+   The classification of `vendor` and `version` will be done using
+   exact match of the values provided by user in profile.
+
+   Investigation details: Below is sample introspection data for firmware.
+   `BIOS` field under firmware have two fields, `vendor` and `version`.
+
+    ```yaml
+    hardware:
+        firmware:
+             bios:
+               date: "10/17/2018"
+               vendor: "Dell Inc."
+               version: "1.5.6"
+    ```
+
+1. SystemVendor
+
+   SystemVendor is new parameter to be introduced which helps user to
+   classify hosts of certain manufacturer and product. User needs to
+   provide `manufacturer` and `productName` in yaml.
```

Comment:
```suggestion
   provide `hardwareCharacteristics.systemVendor.manufacturer` and `hardwareCharacteristics.systemVendor.productName`.
```

Does the user *need* to provide both values? Or can they provide one or the other?